modules/apps/foundation/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/dynamic_inline_scroll.js

```diff
@@ -0,0 +1,161 @@
+AUI.add(
+	'liferay-dynamic-inline-scroll',
+	function(A) {
+		var Lang = A.Lang;
+
+		var TPL_LIST_ITEM = '<li>' +
+				'<a href="{href}" onclick="{onlick}">{pageTitle}</a>' +
+			'</li>';
+
+		var DynamicInlineScroll = A.Component.create(
+			{
+				ATTRS: {
+					cur: {
+						value: 0
+					},
+
+					curParam: {
+						value: ''
+					},
+
+					forcePost: {
+						value: false
+					},
+
+					formName: {
+						value: ''
+					},
+
+					initialPages: {
+						value: 0
+					},
+
+					jsCall: {
+						value: ''
+					},
+
+					namespace: {
+						value: ''
+					},
+
+					pages: {
+						value: 0
+					},
+
+					url: {
+						value: ''
+					},
+
+					urlAnchor: {
+						value: ''
+					}
+				},
+
+				NAME: 'liferaydynamicinlinescroll',
+
+				prototype: {
+					initializer: function() {
+						var instance = this;
+
+						instance._inlineScrollers = A.all('ul.pagination ul.inline-scroller');
+					},
+
+					bindUI: function() {
+						var instance = this;
+
+						instance._inlineScrollers.each(
+							function(item, index) {
+								item.on('scroll', instance._onScroll, instance);
+							}
+						);
+					},
+
+					_getHREF: function(pageIndex) {
+						var instance = this;
+
+						var curParam = instance.get('curParam');
+						var formName = instance.get('formName');
+						var jsCall = instance.get('jsCall');
+						var namespace = instance.get('namespace');
+						var url = instance.get('url');
+						var urlAnchor = instance.get('urlAnchor');
+
+						curParam = namespace + curParam;
+
+						if (url !== null) {
+							return url + curParam + '=' + pageIndex + urlAnchor;
+						}
+
+						return 'javascript:document.' + formName + '.' + curParam + '.value = "' + pageIndex + '"; ' + jsCall;
+					},
+
+					_getOnClick: function(pageIndex) {
+						var instance = this;
+
+						var curParam = instance.get('curParam');
+						var namespace = instance.get('namespace');
+
+						return 'event.preventDefault(); ' + namespace + 'submitForm("' + namespace + curParam + ',' + pageIndex + '");';
+					},
+
+					_onScroll: function(event) {
+						var instance = this;
+
+						var currentPage = Number(instance.get('cur'));
```

Comment:
For the component attributes, you can create a setter (or a getter) for ensuring that the value is always a number. That way you don't have to wrap it in `Number()` every time you need to reference it.